_source/_posts/2023-07-28-scim-workshop.md

```diff
@@ -0,0 +1,1144 @@
+---
+layout: blog_post
+title: "Enterprise-Ready Workshop: SCIM"
+author: semona-igama
+by: advocate
+communities: [security,javascript]
+description: "In this workshop, you will add SCIM support to a sample application, so that user changes made in your app can sync to your customer's Identity Provider!"
+tags: [scim, workshop]
+tweets:
+- ""
+- ""
+- ""
+image: blog/scim-workshop/social.jpg
+type: awareness
+---
+
+Hello SaaS developers! Do you currently have a need to provision, de-provision, and/or sync users across services? If so, then SCIM is the right tool for you! What is SCIM, you ask? SCIM stands for System for Cross-domain Identity Management. And it is an open standard protocol that allows us to manage identities across systems as well as manage common user lifecycles. So in this workshop, our goal is to build a SCIM server in a SaaS app using this standard protocol and, in the end, connect it to an identity provider such as Okta. 
+
+
+In this workshop, we'll cover the following:
+
+{% include toc.md %}
+
+In this workshop, you will learn how to implement SCIM and support provisioning from multiple organizations/tenants. Although we'll integrate with Okta as an example, remember that almost every identity provider on the web has SCIM support! 
+
+# Understanding SCIM 
+
+
+## What problems does SCIM solve?
+There are many challenges with managing user identities across services:
+- SaaS developers want to know as soon as someone joins a company so they can set up an account 
+- SaaS developers also want to know when someone leaves a company to deactivate their account and associated resources.
+- Manually managing users through CSVs
+- Manually adding accounts at scale
+
+SCIM, the System for Cross-domain Identity Management, is an open standard which addresses those problems. SCIM: 
+- Provides near-instant updates to the downstream system whenever someone joins, moves inside of, or leaves a company
+- Allows management of user identities in cloud-based applications and services - i.e., you will have awareness of your users' up-to-date information
```

Comment:
"your users" is too vague, we're talking about the users of the SaaS app developer's enterprise customers here, so make that clear